### PR TITLE
OSX build compatability

### PR DESCRIPTION
**Pull request by curds01**
This represents the work done by @chraibi to get Menge to build in mac.  It includes a couple of final modifications to address some slight code styles.  It replaces PR #35.

### CODE REVIEW

File: projects/g++/Menge/CMakeLists.txt [hunk 1]
```diff
@@ -19,7 +19,8 @@ INCLUDE_DIRECTORIES (${MENGE_SRC_DIR}/MengeCore/Agents)
 INCLUDE_DIRECTORIES (${MENGE_SRC_DIR}/MengeCore/Runtime)
 INCLUDE_DIRECTORIES (${MENGE_SRC_DIR}/MengeCore/PluginEngine)
 
-set( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fpermissive" )
+set( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fpermissive -std=c++11" )
```
Comment (chraibi) [inline projects/g++/Menge/CMakeLists.txt]: this means you don't use compilers without c++11 support. While, this is nice and should be ok, you may still want to check this in the cmakefile.

**Comment by MengeCrowdSim** [comment]
For idiosyncratic historical reasons, Menge was written and developed without C++11 or later.  It's one of my goals to bring it up to date.  One of the big benefits there *may* be to have better synchronization primitives (don't look at the implementation of the readers/writer lock.)  C++11 and 14 provide far better portable constructs for parallelization that I want to include which should allow me to create a legitimate readers/writer lock.  

And then there are other benefits as well.

**Comment by chraibi** [comment]
I understand. But Line 22 above will break if someone is using an old compiler without C++11 support. See [How to detect c++11 support of a compiler with cmake](http://stackoverflow.com/questions/10984442/how-to-detect-c11-support-of-a-compiler-with-cmake)

**Comment by curds01** [comment]
I'm not sure what line you mean by "line 22"; can you provide more detail?

And, regardless of the detail, I have two years of concrete evidence that suggests Menge built just fine without C++11 support.  In fact, what pushed the addition of that support was the change of the `unordered_map` from the `tr1` space to the `std` namespace.

**Comment by chraibi** [comment]
Sean, this is [line 22](https://github.com/MengeCrowdSim/Menge/blob/master/projects/g%2B%2B/Menge/CMakeLists.txt#L22). It assumes the user is using a compiler with C++11 support, which may not be the case.

**Comment by curds01** [comment]
Aaaaah, I understand now.  Your concern is correct, but I feel the danger is slim to none.  The risk of someone using a compile that doesn't support this flag is slight, and decreases every day.

**Comment by chraibi** [comment]
😃

**Pull request merged**